052-record-reconciled-versions.md

```diff
@@ -0,0 +1,197 @@
+
+Same as fresh install, upgrade from 0.37.0 to 0.38.0 with Kafka cersion 3.5.1
+- On an upgrade, a user has a Kafka CR:
+  ```
+  # the status.versions is not set, but treat it as empty here
+  status: {}
+  ```
+- The KafkaReconciler reconciles the CR and creates `StrimziPodSet` CRs
+- StrimziPodSet reconciler reconciles the `SPS`s creating the Kafka Brokers
+- Kafka Reconciler picks up that the Kafka Brokers managed by the SPS are now Ready and finished installing.
+  Updating the status with the reconciled kafka versions:
+  ```
+  status:
+    versions:
+      kafka: 3.5.1
+  ```
+- Kafka Reconciler continues to install other components such as entity operator.
+  Once all components are deployed and updated, update `Kafka` CR with
+  ```
+  status:
+    versions:
+      lastSuccessfulReconciliationBy: 0.38.0
+      kafka: 3.5.1
+  ```
+- User can now verify `Kafka.status.versions.lastSuccessfulReconciliationBy` and `Kafka.status.versions.kafka` to check if it has reconciled to correct/expected levels.
+- if there is an error at any point prior to the Kafka Broker rollout `Kafka.status.versions.kafka` will not be updated.
+- if there is an error at any point during reconcile `Kafka.status.versions.lastSuccessfulReconciliationBy` will not be updated.
+
+
+### Upgrade (with new mechansim)
+
+This defines an example where it is imagined that `0.37.0` had this mechanism already implemented, and is now upgrading to a new version `0.38.0` (with kafka upgrading from `3.5.0` to `3.5.1`)
+
+- On an upgrade, a user has a Kafka CR:
+  ```
+  status:
+    versions:
+      lastSuccessfulReconciliationBy: 0.37.0
+      kafka: 3.5.0
+  ```
+- The KafkaReconciler reconciles the CR and creates `StrimziPodSet` CRs
+- StrimziPodSet reconciler reconciles the `SPS`s creating the Kafka Brokers
+- Kafka Reconciler picks up that the Kafka Brokers managed by the SPS are now Ready and finished installing.
+  Updating the status with the reconciled kafka versions:
+  ```
+  status:
+    versions:
+      lastSuccessfulReconciliationBy: 0.37.0
+      kafka: 3.5.1
+  ```
+- Kafka Reconciler continues to install other components such as entity operator.
+  Once all components are deployed and updated, update `Kafka` CR with
+  ```
+  status:
+    versions:
+      lastSuccessfulReconciliationBy: 0.38.0
+      kafka: 3.5.1
+  ```
+- User can now verify `Kafka.status.versions.lastSuccessfulReconciliationBy` and `Kafka.status.versions.kafka` to check if it has reconciled to correct/expected levels.
+- if there is an error at any point prior to the Kafka Broker rollout `Kafka.status.versions.kafka` will not be updated.
+- if there is an error at any point during reconcile `Kafka.status.versions.lastSuccessfulReconciliationBy` will not be updated.
+
+## `KafkaNodePool` considerations
+
+Some users use the new `KafkaNodePool` CRs to facilitate upgrades through rolling out new brokers with new Kafka versions, reassigning the partitions and then delete the old brokers.
```

Comment:
It is not that someone uses it today as this might suggest. It is more something to consider. So maybe this would be better wording?

```suggestion
It might happen that in the future each `KafkaNodePool` CR supports having its own Kafka version which might differ from other `KafkaNodePools` in the same Kafka cluster.
```